distributed/tests/test_worker.py

```diff
@@ -1402,3 +1402,25 @@ def test_host_address(c, s):
     assert "127.0.0.3" in n.address
     assert "127.0.0.3" in n.worker_address
     yield n.close()
+
+
+def test_resource_limit():
+    assert parse_memory_limit("250MiB", 1, total_cores=1) == 1024 * 1024 * 250
+
+    child = os.fork()
+    if child == 0:
+        # set lower memory limit for the child process (RSS limit doesn't
+        # affect the parent process)
+        try:
+            import resource
+
+            limit = 1024 * 1024 * 200
+            resource.setrlimit(resource.RLIMIT_RSS, (limit, limit))
+            if parse_memory_limit("250MiB", 1, total_cores=1) != limit:
+                os._exit(1)
+        except (ImportError, OSError):
+            pass
+        os._exit(0)
+    else:
+        # wait for child
+        assert os.waitpid(child, 0)[1] == 0
```

Comment:
I have never used this API.  It was interesting to see it work.  Seems pretty slick to me.  

Our window CI currently has some issues.  I'm trying to fix it up today.  Hopefully I'll be able to rerun tests afterwards to make sure that things here pass well there.  If not then we'll have to skip some behavior based on OS.  I'll update here when that's done.

Reply:
I also used it for the first time and was surprised how simple it is :)

Is it possible from within a test function to mark it as skipped? If `resource` is not found or an `OSError` occurs, it might be nice to signal that the test had to be skipped.

Reply:
Yes, you can decorate the function with 

```python
@pytest.mark.skipif(sys.platform.startswith('win'), reason="Fork not available")
def test_foo():
```

Or you can call `pytest.skip("a reason")` from within the function. 

Judging by the windows tests (which work again) `os.fork` isn't available so we'll need to skip this.

Maybe

```python
@pytest.mark.skipif(not hasattr(os, 'fork'), reason="Fork not available")
def test_foo():
```

Reply:
thank you! Lets see whether it works now.

It would probably be better to have a test case which also works on windows! I assume that would either mean to open a new python instance with subprocess or to simply change the RSS limit for the current process. I think it would be safe to decrease the limit to maximal memory -1 (if no limit is set) or current limit -1.